File: src/main/java/com/rpb/reservation/reservation/application/PlatformReservationShareTemplateSeed.java
package com.rpb.reservation.reservation.application;

import java.util.List;

public record PlatformReservationShareTemplateSeed(
    String seedKey,
    String displayName,
    String locale,
    String templateText,
    String status,
    int version,
    List<String> allowedVariables
) {
    public PlatformReservationShareTemplateSeed {
        allowedVariables = List.copyOf(allowedVariables == null ? List.of() : allowedVariables);
    }
}

File: src/types/platformReservationShareTemplateSeed.ts
export type PlatformReservationShareTemplateSeedStatus = 'active' | 'disabled'

export interface PlatformReservationShareTemplateSeed {
  seedKey: string
  displayName: string
  locale: string
  templateText: string
  status: PlatformReservationShareTemplateSeedStatus
  version: number
  allowedVariables: string[]
}

export interface PlatformReservationShareTemplateSeedResponse {
  success: true
  seed: PlatformReservationShareTemplateSeed
}

export interface PlatformReservationShareTemplateSeedMutation {
  displayName: string
  locale: string
  templateText: string
  status: PlatformReservationShareTemplateSeedStatus
  version?: number
}

export interface PlatformReservationShareTemplateSeedApiErrorResponse {
  success: false
  error: {
    code: string
    messageKey: string
    details: Record<string, unknown>
  }
}
